File: 006_ZigZag_Conversion/solution.go
func convert(s string, numRows int) string {
    if numRows == 1 {
        return s
    }
    runes := []rune(s)
    rows := make([][]rune, numRows)
    i := 0
    step := 1
    for index := range runes {
        rows[i] = append(rows[i], runes[index])
        if i == numRows - 1 {
            step = -1
        } else if i == 0 {
            step = 1
        }
        i += step
    }
    retRunes := []rune{}
    for index := range rows {
        retRunes = append(retRunes, rows[index]...)
    }
    return string(retRunes)
}

File: 006_ZigZag_Conversion/solution.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1:
            return s
        tmp = ['' for i in range(numRows)]
        index = -1
        step = 1
        for i in range(len(s)):
            index += step
            if index == numRows:
                index -= 2
                step = -1
            elif index == -1:
                index = 1
                step = 1
            tmp[index] += s[i]
        return ''.join(tmp)
